src/main/java/org/synyx/urlaubsverwaltung/mail/MailServiceImpl.java

```diff
@@ -28,92 +29,83 @@
     private final MailBuilder mailBuilder;
     private final MailSenderService mailSenderService;
     private final MailProperties mailProperties;
-    private final RecipientService recipientService;
+    private final PersonService personService;
 
     @Autowired
     MailServiceImpl(MessageSource messageSource, MailBuilder mailBuilder, MailSenderService mailSenderService,
-                    MailProperties mailProperties, RecipientService recipientService) {
+                    MailProperties mailProperties, PersonService personService) {
 
         this.messageSource = messageSource;
         this.mailBuilder = mailBuilder;
         this.mailProperties = mailProperties;
         this.mailSenderService = mailSenderService;
-        this.recipientService = recipientService;
-    }
-
-    @Override
-    public void sendMailTo(MailNotification mailNotification, String subjectMessageKey, String templateName, Map<String, Object> model) {
-
-        final List<Person> persons = recipientService.getRecipientsWithNotificationType(mailNotification);
-        sendMailToPersons(persons, subjectMessageKey, templateName, model);
-    }
-
-    @Override
-    public void sendMailTo(Person person, String subjectMessageKey, String templateName, Map<String, Object> model) {
-
-        final List<Person> persons = singletonList(person);
-        sendMailToPersons(persons, subjectMessageKey, templateName, model);
-    }
-
-    @Override
-    public void sendMailTo(Person person, String subjectMessageKey, String templateName, Map<String, Object> model, List<Attachment> attachments) {
-
-        final List<Person> persons = singletonList(person);
-        sendMailToPersons(persons, subjectMessageKey, templateName, model, attachments);
+        this.personService = personService;
     }
 
     @Override
-    public void sendMailToEach(List<Person> persons, String subjectMessageKey, String templateName, Map<String, Object> model, Object... args) {
+    public void send(Mail mail) {
 
-        persons.forEach(person -> {
-            model.put("recipient", person);
-            final List<String> mailAddress = recipientService.getMailAddresses(person);
-            sendMailToRecipients(mailAddress, subjectMessageKey, templateName, model, args);
-        });
-    }
+        final Map<String, Object> model = mail.getTemplateModel();
+        model.put("baseLinkURL", getApplicationUrl());
 
-    @Override
-    public void sendTechnicalMail(String subjectMessageKey, String templateName, Map<String, Object> model) {
-        sendMailToRecipients(singletonList(mailProperties.getAdministrator()), subjectMessageKey, templateName, model);
+        final String subject = getTranslation(mail.getSubjectMessageKey(), mail.getSubjectMessageArguments());
+        final String sender = mailProperties.getSender();
+
+        if (mail.isSendToTechnicalMail()) {
```

Comment:
`if/else` does not ensure that current builder pattern allows both `sendToTechnical` **and** to other recipients

e.g.
```
Mail.builder()
            .withRecipient(NOTIFICATION_OFFICE)
            .withRecipient(true)
            .withSubject(subjectMessageKey)
            .withTemplate(templateName, model)
            .build();
```

Reply:
Yes thats right at was on purpose, the idea was to get rid of the technical emails and then we only have the ugly if else part that I wanted to restructure before merging this pr. I have to think about it.